### PR TITLE
Cleanup title and comments

### PR DESCRIPTION
**Pull request by ayush1999**
Initial changes : Added a small change for replacing all links with URL token. Will add more changes upon approval.

### CODE REVIEW

File: bugbug/bug_features.py [hunk 1]
```diff
@@ -179,6 +179,9 @@ def __init__(self, feature_extractors, commit_messages_map=None):
     def fit(self, x, y=None):
         return self
 
+    def cleanup(self, text):
+        return re.sub(r'http\S+', 'URL', text)
```
Comment (marco-c) [inline bugbug/bug_features.py]: This will also match invalid URLs, but it should be fine.

Comment (marco-c) [reply bugbug/bug_features.py]: I'd define, outside of this class, a list of "text cleanup" classes (just like the classes we have for feature extraction). Then you call them either in this `cleanup` function, or in `transform`.

File: bugbug/models/bug.py [hunk 2]
```diff
@@ -36,6 +36,7 @@ def __init__(self, lemmatization=False):
             bug_features.landings(),
             bug_features.title(),
             bug_features.comments(),
+            bug_features.cleanup_url(),
```
Comment (marco-c) [inline bugbug/models/bug.py]: Sorry for the confusion, I didn't mean to make it another feature extractor. I just meant we should have "cleanup" functions implemented similarly to feature extractors.

Comment (marco-c) [reply bugbug/models/bug.py]: So, call the cleanup functions in the place where you were calling it before, but making it more generic.

Comment (marco-c) [reply bugbug/models/bug.py]: Something like:
```
for cleanup_function in cleanup_functions:
    summary = cleanup_function(summary)
    comments = cleanup_function(comments)
```

**Comment by marco-c** [comment]
> @marco-c I've created a new class that replaces all URLs with the token. Should I do so for remaining changes? (stack traces, file references etc)

Let's add one cleanup function per PR. After you add it, please also comment in the PR with the results of the training before/after your change (with the 'bug' and 'regression' models).

**Comment by ayush1999** [comment]
@marco-c Current results:
Before changes:
```
ayush99@ayush99:~/GitHub/bugbug$ python run.py --train --goal=bug
(1627, 101401) (1627,)
(181, 101401) (181,)
CV Accuracy: 0.89 (+/- 0.05)
Accuracy: 0.9171270718232044
Precision: 0.9642857142857143
Recall: 0.8709677419354839
[[85  3]
 [12 81]]
```
After changes:
```
ayush99@ayush99:~/GitHub/bugbug$ python run.py --train --goal=bug
(1627, 85316) (1627,)
(181, 85316) (181,)
CV Accuracy: 0.90 (+/- 0.04)
Accuracy: 0.9171270718232044
Precision: 0.9642857142857143
Recall: 0.8709677419354839
[[85  3]
 [12 81]]
```

**Comment by ayush1999** [comment]
In case of regression, results are exactly same.

File: bugbug/bug_features.py [hunk 3]
```diff
@@ -205,11 +210,12 @@ def transform(self, bugs):
 
             # TODO: Try simply using all possible fields instead of extracting features manually.
 
-            result = {
-                'data': data,
-                'title': bug['summary'],
-                'comments': ' '.join([c['text'] for c in bug['comments']]),
-            }
+            for cleanup_function in self.cleanup_functions:
+                result = {
```
Comment (marco-c) [inline bugbug/bug_features.py]: Thanks, the patch is looking better!

A couple of things to fix:
1. The cleanup functions should be applied to both the summary and the comments;
2. The result should not be redefined multiple times, but just once. In the loop you should just apply the cleanup functions to the textual fields.

**Pull request merged**